### PR TITLE
Add CCR Follow Info API

### PR DESCRIPTION
**Pull request by codebrain**
Add ccr follow info API 

https://github.com/elastic/elasticsearch/issues/37408

### CODE REVIEW

File: src/Nest/XPack/CrossClusterReplication/Follow/FollowInfo/FollowConfig.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		public int MaximumWriteBufferCount { get; internal set; }
+
+		[DataMember(Name = "max_write_buffer_size")]
+		public string MaximumWriteBufferSize { get; internal set; }
+
+		[DataMember(Name = "max_retry_delay")]
+		public TimeSpan MaximumRetryDelay { get; internal set; }
+
+		[DataMember(Name = "read_poll_timeout")]
+		public TimeSpan ReadPollTimeout { get; internal set; }
```
Comment (russcam) [inline src/Nest/XPack/CrossClusterReplication/Follow/FollowInfo/FollowConfig.cs]: Should be `Time` type

File: src/Nest/XPack/CrossClusterReplication/Follow/FollowInfo/FollowConfig.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		public int MaximumOutstandingWriteRequests { get; internal set; }
+
+		[DataMember(Name = "max_write_buffer_count")]
+		public int MaximumWriteBufferCount { get; internal set; }
+
+		[DataMember(Name = "max_write_buffer_size")]
+		public string MaximumWriteBufferSize { get; internal set; }
+
+		[DataMember(Name = "max_retry_delay")]
+		public TimeSpan MaximumRetryDelay { get; internal set; }
```
Comment (russcam) [inline src/Nest/XPack/CrossClusterReplication/Follow/FollowInfo/FollowConfig.cs]: Should be `Time` type

File: src/Tests/Tests/XPack/CrossClusterReplication/CrossClusterReplicationFollowTests.cs [hunk 3]
```diff
@@ -271,6 +283,11 @@ public CrossClusterReplicationFollowTests(XPackCluster cluster, EndpointUsage us
 			}
 		});
 
+		[I] public async Task FollowInfoResponse() => await Assert<FollowInfoResponse>(FollowInfoStep, r =>
+		{
+			r.IsValid.Should().BeTrue();
```
Comment (Mpdreamz) [inline src/Tests/Tests/XPack/CrossClusterReplication/CrossClusterReplicationFollowTests.cs]: Add some more assertions on the actual response.

**Comment by Mpdreamz** [comment]
One unit test failure:
```
Tests.CodeStandards.Serialization.Enums.EnumsWithEnumMembersShouldBeMarkedWithStringEnumAttribute [FAIL]
Failed   Tests.CodeStandards.Serialization.Enums.EnumsWithEnumMembersShouldBeMarkedWithStringEnumAttribute
Error Message:
 Expected collection to be empty, but found {"FollowerIndexStatus"}.
Stack Trace:
```

**Pull request merged**